src/uwtools/drivers/ungrib.py

```diff
@@ -0,0 +1,175 @@
+"""
+A driver for the ungrib component.
+"""
+
+import os
+import stat
+from datetime import datetime
+from pathlib import Path
+from typing import Any, Dict
+
+from iotaa import asset, dryrun, task, tasks
+
+from uwtools.config.formats.nml import NMLConfig
+from uwtools.drivers.driver import Driver
+from uwtools.strings import STR
+from uwtools.utils.tasks import file
+
+
+class Ungrib(Driver):
+    """
+    A driver for ungrib.
+    """
+
+    _driver_name = STR.ungrib
+
+    def __init__(
+        self, config_file: Path, cycle: datetime, dry_run: bool = False, batch: bool = False
+    ):
+        """
+        The driver.
+
+        :param config_file: Path to config file.
+        :param cycle: The forecast cycle.
+        :param dry_run: Run in dry-run mode?
+        :param batch: Run component via the batch system?
+        """
+        super().__init__(config_file=config_file, dry_run=dry_run, batch=batch)
+        self._config.dereference(context={"cycle": cycle})
+        if self._dry_run:
+            dryrun()
+        self._cycle = cycle
+
+    # Workflow tasks
+
+    @task
+    def gribfile_aaa(self):
+        """
+        The gribfile.
+        """
+        path = self._rundir / "GRIBFILE.AAA"
+        yield self._taskname(str(path))
+        yield asset(path, path.is_symlink)
+        infile = Path(self._driver_config["gfs_file"])
+        yield file(path=infile)
+        path.parent.mkdir(parents=True, exist_ok=True)
+        path.symlink_to(infile)
```

Comment:
This will definitely work for preparing initial conditions. I think (and I might be wrong) that you can do a whole batch of GFS files to handle all the times for ICs and LBCs in one iteration. In that case links would look something like:

```
GRIBFILE.AAA -> "/path/to/dir/gfs.t12z.pgrb2.0p25.f000"
GRIBFILE.AAB -> "/path/to/dir/gfs.t12z.pgrb2.0p25.f006"
GRIBFILE.AAC -> "/path/to/dir/gfs.t12z.pgrb2.0p25.f012"
```

This does not need to be modified now, but would need to be extended for limited-area in this way at some point if I'm remembering correctly from my WRF days.